server/src/main/java/io/spine/server/aggregate/IdempotencyGuard.java

```diff
@@ -45,13 +45,13 @@
     }
 
     /**
-     * Ensures that the command was not dispatched to the aggregate.
+     * Checks that the command was not dispatched to the aggregate.
      * If it was a {@link DuplicateCommandException} is thrown.
      *
      * @param envelope an envelope with a command to check
      * @throws DuplicateCommandException if the command was dispatched to the aggregate
      */
-    void ensureIdempotence(CommandEnvelope envelope) {
+    void check(CommandEnvelope envelope) {
```

Comment:
Can we delegate the check to the `RecentHistory` class instead? I mean the history can answer on the question: “Did we have events that were originated with this command?” If yes, then the guard would fire the exception.

Reply:
It sure can. 

But Recent History is just an aggregates `historyBackwards()` implementation detail for now.
Adding the `containsEventsCausedBy(Command)` to the aggregate would be bloating its API. There’s no other case for using this as an aggregate method.